src/app/(proper_react)/redesign/(authenticated)/user/dashboard/page.tsx

```diff
@@ -33,7 +40,45 @@ export default async function DashboardPage() {
     return redirect("/redesign/user/welcome/");
   }
 
+  // This contains the latest scan results in our database.
   const latestScan = await getLatestOnerepScanResults(profileId);
+
+  // Attempt to fetch the current scan results from the provider.
```

Comment:
Might it make sense to extract this into a `refreshStoredScanResults` function? Since I assume the same code needs to run in response to webhook calls? (Or does the Webhook call already include the latest data?)

Reply:
The webhook payload only includes the scan ID that needs update, sadly :/ I'll see if there's any opportunity for more code sharing though: https://github.com/mozilla/blurts-server/blob/main/src/app/api/v1/onerep-events/route.ts

Reply:
OK I extracted this to a server function and out of the dashboard component for readability - I think it could be used from the onerep-events webhook potentially but there are some differences:

1. the webhook includes the scan ID as well as the profile ID, and we don't try to filter duplicates
2. `refreshStoredScanResults` as-is just refreshes everything for a give profile ID, filtering duplicates

We could have one function that could do both, but they don't have that much in common except for a few function calls to functions in the `onerep.ts` module.